<?xml version="1.0"?>
<launch>

    <!-- For Velodyne HDL-64 -->
    <param name="scan_line" value="16" />
    
    
    <!--- Sim Time -->
    <param name="/use_sim_time" value="true" />
    <param name="scan_period" value="0.1" />

    <param name="vertical_angle" type="double" value="2.0" />
    <param name="max_dis" type="double" value="90.0" />
    <param name="map_resolution" type="double" value="0.4" />
    <param name="min_dis" type="double" value="3.0" />
    <!--- -->
    <param name="laser_topic" type="string" value="/velodyne_points"/>
    <param name="edge_resolution" type="double" value="0.2" />
    <param name="surf_resolution" type="double" value="0.5" />
    <param name="sampling_rate" type="double" value="2.0" />
    <param name="min_noise_prior" type="double" value="0.02" />
    <param name="edge_noise_threshold" type="double" value="0.1" />
    <param name="plane_noise_threshold" type="double" value="0.05" />
    <param name="cube_len" type="double" value="800" />
    <node pkg="floam" type="floam_laser_odom_node" name="floam_laser_odom_node" output="screen"/>
    <!-- <node pkg="floam" type="floam_feature_extract_node" name="floam_feature_extract_node" output="screen"/> -->
     <node pkg="floam" type="floam_laser_processing_node" name="floam_laser_processing_node" output="screen"/>
    <node pkg="floam" type="floam_laser_mapping_node" name="floam_laser_mapping_node" output="screen"/>

    <node pkg="tf" type="static_transform_publisher" name="word2map_tf"  args="0 0 0 0 0 0 /world /map 10" />
    
    <arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find floam)/rviz/test.rviz" />
    </group>

    
  	<node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_loam" ns="gt" >
        <param name="/target_frame_name" value="world" />
        <param name="/source_frame_name" value="velodyne" />
        <param name="/trajectory_update_rate" value="10.0" />
        <param name="/trajectory_publish_rate" value="10.0" />
    </node>
    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_loam" ns="base_link" >
        <param name="/target_frame_name" value="world" />
        <param name="/source_frame_name" value="base_link" />
        <param name="/trajectory_update_rate" value="10.0" />
        <param name="/trajectory_publish_rate" value="10.0" />
    </node>

</launch>
